src/Punycode.php

```diff
@@ -59,11 +59,31 @@ class Punycode
      * Constructor
      *
      * @param string $encoding Character encoding
+     * @throws \RuntimeException
      */
     public function __construct($encoding = 'UTF-8')
     {
+        if (false === $this->isEncodingSupported($baseEncoding)) {
+            throw new \RuntimeException("Encoding $encoding is not supported");
+        }
         $this->encoding = $encoding;
     }
+    
+    /**
+     * Check if provided encoding is supported by mb_string
+     *
+     * @param string $checkedEncoding
+     * @return bool
+     */
+    private function isEncodingSupported($checkedEncoding)
+    {
+        foreach (mb_list_encodings() as $encoding) {
```

Comment:
Encoding can also have a number of aliases, as returned by `mb_encoding_aliases`, should we check this method too or the `mb_list_encodings` include the aliases too?